encoder/libav_encoder.cpp

```diff
@@ -723,30 +799,13 @@ void LibAvEncoder::audioThread()
 			AVRational num = { 1, out_frame->sample_rate };
 			int64_t ts = av_rescale_q(audio_samples_, num, codec_ctx_[AudioOut]->time_base);
 
-			// Need to set up timestamps based on the video recording settings:
-			if (previous_audio_timestamp_ == 0)
-				previous_audio_timestamp_ = ts; // Only useful on startup
-
-			int64_t et = ts - previous_audio_timestamp_; // Elapsed Time
+			out_frame->pts = ts + (options_->av_sync < 0 ? -options_->av_sync : 0);
```

Comment:
Have a look at https://github.com/raspberrypi/libcamera-apps/commit/f5fe27124531af19f68dd8ead18a05773c0f1db3.

I think this wants to be
```
out_frame->pts = ts + (options_->av_sync > 0 ? options_->av_sync : 0);
```